### ISSUE TITLE
[Feature]: Add Arrow Flight SQL 

### ISSUE THREAD

**Issue by YuriyGavrilov** [oceanbase/oceanbase#1632]
**Use Case**
Retrieve data faster through arrow flight driver 

**Describe the solution you'd like**
see above. 

**Describe alternatives you've considered**
Actually there is no. 

**Additional context**
There is new modern SQL way to retrieve data in a faster way through Arrow Flight SQL driver.
https://arrow.apache.org/docs/java/flight_sql_jdbc_driver.html#
https://arrow.apache.org/blog/2022/02/16/introducing-arrow-flight-sql/
https://www.postgresql.org/about/news/apache-arrow-flight-sql-adapter-for-postgresql-010-2716/

Some explanations and research:
https://www.vldb.org/pvldb/vol10/p1022-muehleisen.pdf

Some server examples:
https://github.com/voltrondata/flight-sql-server-example
https://www.kamu.dev/blog/2023-09-datafusion-flightsql/ https://github.com/kamu-data/kamu-cli

**Comment by caifeizhi** [comment]
Thank you for the suggestion. I have a few concerns here. 

1. Is this a faster JDBC driver, Can it be compared to JDBC? 
2. Can the application seamlessly switch to it?
3. How to (or Where can we ) "Add" Add Arrow Flight SQL?

**Comment by YuriyGavrilov** [comment]
> Thank you for the suggestion. I have a few concerns here.
> 
> 1. Is this a faster JDBC driver, Can it be compared to JDBC?
> 2. Can the application seamlessly switch to it?
> 3. How to (or Where can we ) "Add" Add Arrow Flight SQL?

Hi @caifeizhi 

Sure, here some info:

1 . for example some comparisons
https://voltrondata.com/resources/comparing-performance-of-adbc-and-jdbc-in-python-for-arrow-flight-sql
https://youtu.be/dQszohqgZbQ?si=qT7GhNZE4XHCY_V_
https://arrow.apache.org/blog/2019/10/13/introducing-arrow-flight/

2. Yes, Arrow Driver JDBC driver publish regular.
https://central.sonatype.com/search?q=a:flight-sql-jdbc-driver&smo=true
There is 14.x version at this moment. 

3. The best way is to do it like other tool do it already. They just add additional port to their tools and open it to receive\or send data over flight  SQL protocol. For example Ocean uses 2881 port as default and 1882 as arrow flight sql protocol.

**Comment by caifeizhi** [comment]
Thank you very much for your suggestion. Here is the conclusion I understand (please correct me if I'm wrong):
1. Arrow-flight-sql adopts a new interactive protocol that supports high concurrency and distributed data reading, and has better performance compared to traditional JDBC；
2. Introducing arrow-flight-sql requires support for the arrow-flight protocol in the database backend, and modification of the JDBC connection method at the application layer (i.e., the client).

**Comment by YuriyGavrilov** [comment]
@caifeizhi yep, it is correct. Just with to add that on client side modifications are minimal. Sometimes it is just to change driver type.

**Comment by caifeizhi** [comment]
If we want to adopt this solution, what needs to be done at the database backend?

I found pg supported with extension https://github.com/apache/arrow-flight-sql-postgresql , but, unfortunately, OceanBase does not support a extension system.

**Comment by YuriyGavrilov** [comment]
@caifeizhi I think better to start own solution instead of using Postgres extension. For example it could be like in this solution  https://github.com/voltrondata/flight-sql-server-example/tree/main/src
